index.js

```diff
@@ -101,12 +112,37 @@ class Client {
       this.accessToken = tokens.access
       this.refreshToken = tokens.refresh
 
-      uuid = await requester.do(options, this.accessToken, this.apiUrl)
+      requestResponse = await requester.do(options, this.accessToken, this.apiUrl)
+    }
+
+    const uuid = requestResponse.uuid
+    let timeout = options.timeout
+
+    // FIXME: this might not be optimal. The test should be negated
+    // and then do the *only* subset of poller that needs to be done,
+    // given that we know this is finished. Instead, I think we'll
+    // make one more additional request of status only to find out
+    // that it is still finished.
```

Comment:
Correct, if status is `Finished` at this point, there is no need to run entire poller logic, it is enough just to execute this piece directly: https://github.com/ConsenSys/armlet/blob/master/lib/poller.js#L71-L78

I'd say, split that piece into a separate `getIssues(..)` function inside `poller.js`, that will be called from `ping(..)`, and also exported and directly called here.

Reply:
Yeah, that is what I tried to do unsuccessfully the other day and couldn't get it to work.

Reply:
Do you need help with it?

Reply:
YES! If you would do it, I'd be grateful.